### PR TITLE
(maint) port tests to cloud ci

### PR DESCRIPTION
**Pull request by sheenaajay**
Request for a review
Added plans for provisioning and puppetserver setup for cloud ci
Modified spec_helper_acceptance_local for puppet server/agents configuration and replaced with docker tool to generate the certs
Added tests to verify controller/worker.
Added new tags for testing integration in Rakefile
Added tags to the tests

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -2,15 +2,19 @@ fixtures:
...
     stdlib: 'puppetlabs-stdlib'
     apt: 'puppetlabs-apt'
+    puppet_conf: 'puppetlabs-puppet_conf'
     archive: "puppet-archive"
     translate: "puppetlabs-translate"
     augeasproviders_sysctl: "herculesteam-augeasproviders_sysctl"
     augeasproviders_core: "herculesteam-augeasproviders_core"
     kmod: "camptocamp-kmod"
+    helm: 'puppetlabs-helm'
+    rook: 'puppetlabs-rook'
```
Comment (carabasdaniel) [inline .fixtures.yml:12]: you don't need to add rook in the forge_modules if you have it set in the repositories

File: .github/workflows/integration_test.yml [hunk 2]
```diff
@@ -110,33 +103,41 @@ jobs:
...
         echo ::group::=== REQUEST ===
         cat request.json || true
         echo
         echo ::endgroup::
         echo ::group::=== INVENTORY ===
         sed -e 's/password: .*/password: "[redacted]"/' < inventory.yaml || true
         echo ::endgroup::
-
-    - name: Install agent
+    - name: Install module
```
Comment (carabasdaniel) [inline .github/workflows/integration_test.yml:127]: This should probably be the puppetserver setup step

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -112,33 +105,41 @@ jobs:
...
         echo ::group::=== REQUEST ===
         cat request.json || true
         echo
         echo ::endgroup::
         echo ::group::=== INVENTORY ===
         sed -e 's/password: .*/password: "[redacted]"/' < inventory.yaml || true
         echo ::endgroup::
-
-    - name: Install agent
+    - name: Install module
```
Comment (carabasdaniel) [inline .github/workflows/nightly.yml:129]: this should be the puppetserver setup step

File: plans/provision_gcp.pp [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# @summary Provisions machines
+#
+# Provisions machines for integration testing
+#
+# @example
+#   kubernetes::provision_integration
+plan kubernetes::provision_gcp(
```
Comment (carabasdaniel) [inline plans/provision_gcp.pp:7]: this should probably be renamed to provision_cluster

File: spec/acceptance/kubernetes_spec.rb [hunk 5]
```diff
@@ -48,17 +49,17 @@ class {'kubernetes':
...
           run_shell('KUBECONFIG=/etc/kubernetes/admin.conf kubectl create -f /tmp/nginx.yml') do |r|
             expect(r.stdout).to match(/my-nginx created\nservice\/my-nginx created\n/)
           end
         end
 
         it 'can access the deployed service' do
-          run_shell('sleep 60')
-          run_shell('curl -s 10.96.188.5') do |r|
+          sleep(180)
+          run_shell('curl -s 10.96.188.5', expect_failures: true) do |r|
```
Comment (carabasdaniel) [inline spec/acceptance/kubernetes_spec.rb:62]: if you have a couple of failures with this call and you might want to add a retry mechanism here. see the --retry options for curl

**Comment by sheenaajay** [comment]
Thanks @carabasdaniel for the valuable comments.

**Review by carabasdaniel — APPROVED** [review]

**Pull request merged**